### PR TITLE
fixes for ruff warning C416

### PR DESCRIPTION
**Pull request by fchapoton**
this fixes almost every C416 warning from `ruff`

Unnecessary `list` comprehension (rewrite using `list()`)

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/rings/function_field/order_polymod.py [hunk 1]
```diff
@@ -1325,8 +1325,8 @@ def bar(I): # transfer to O/pO
 
             def liftb(Ib):
                 m = [vector([fr(e) for e in v]) for v in Ib]
-                m += [v for v in pO._hnf]
-                return self._ideal_from_vectors_and_denominator(m,1)
+                m.append(list(pO._hnf))
```
Comment (dcoudert) [inline src/sage/rings/function_field/order_polymod.py]: Are you sure it's the right form ? I think it should be `m.extend(pO._hnf)`.

**Review by dcoudert — CHANGES_REQUESTED** [review]
otherwise, LGTM.

File: src/sage/rings/function_field/order_polymod.py (hunk 1 above)
Comment (fchapoton) [reply src/sage/rings/function_field/order_polymod.py]: oops, oui, désolé

**Review by dcoudert — APPROVED** [review]
LGTM.

**Pull request merged**